application/app/models/event.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class Event
+  include ActiveModel::Model
+  include YamlStorageCommon
+
+  ENTITY_TYPES = %w[project download_file upload_bundle upload_file].freeze
+  ATTRIBUTES = %w[id project_id message entity_type entity_id creation_date metadata].freeze
+
+  attr_accessor(*ATTRIBUTES)
+
+  validates_presence_of :id, :project_id, :message, :entity_type, :creation_date
+
+  def initialize(attributes = {})
+    super
+    self.id ||= SecureRandom.uuid.to_s
```

Comment:
for this sort of model, I would not allow the id and the date to be set outside this class.
As well, this class should be immutable.

Reply:
I can make id to be only set internally in this class. However, for the date it may be interesting to make entity creation date and the event date exactly match. This is why I did it this way and in fact I force in the projects controller to have the same creation date on project create

Reply:
I changed both attributes